packages/shared_preferences/shared_preferences/lib/shared_preferences.dart

```diff
@@ -161,9 +208,27 @@ class SharedPreferences {
   }
 
   static Future<Map<String, Object>> _getSharedPreferencesMap() async {
-    final Map<String, Object> fromSystem = await _store.getAll();
-    assert(fromSystem != null);
-    // Strip the flutter. prefix from the returned preferences.
+    final Map<String, Object> fromSystem = <String, Object>{};
+    if (_prefixHasBeenChanged) {
+      try {
+        fromSystem.addAll(await _store.getAllWithPrefix(_prefix));
+      } catch (e) {
+        // Catching and clarifying UnimplementedError to provide a more robust message.
+        if (e is UnimplementedError) {
+          throw UnimplementedError('''
+This implementation of Shared Preferences doesn't yet support the setPrefix method.
+Either update the implementation to support setPrefix, or do not call setPrefix.
+        ''');
+        }
```

Comment:
Same

Reply:
Since this has had two different bugs, let's actually add tests that if the platform implementation throws a `PlatformException` it's passed correctly through both of these methods (by making a fake platform implementation that always throws, and setting it as the instance in those tests).